src/core/Expander/Expander/Expander.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { Component, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import { AutoId } from '../../../utils/AutoId';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { HtmlDiv } from '../../../reset';
+import { baseStyles } from './Expander.baseStyles';
+import {
+  ExpanderGroupConsumer,
+  ExpanderGroupProviderState,
+} from '../ExpanderGroup/ExpanderGroup';
+
+const baseClassName = 'fi-expander';
+const openClassName = `${baseClassName}--open`;
+
+export interface ExpanderProviderState {
+  /** Callback for communicating ExpanderTitle button event to Expander  */
+  onToggleExpander: () => void;
+  /** Open state for expander */
+  open: boolean;
+  /** Id for expander button */
+  titleId: string | undefined;
+  /** Id for expander content */
+  contentId: string | undefined;
+}
+
+const defaultProviderValue: ExpanderProviderState = {
+  onToggleExpander: () => null,
+  open: false,
+  titleId: undefined,
+  contentId: undefined,
+};
+
+const {
+  Provider: ExpanderProvider,
+  Consumer: ExpanderConsumer,
+} = React.createContext(defaultProviderValue);
+
+interface InternalExpanderProps {
+  /**
+   * Children, extend type ExpanderTitleBaseProps or ExpanderContentBaseProps
+   * ExpanderProviderState context is used to communicate between title, content and expander
+   */
```

Comment:
**Not to change anything regarding this PR**

Just to add note for future, if it is ever going to be supported in JSDoc, support for e.g `@see`.
So that these comments that will refer to these props/interfaces are not just text, but actual "link".
Something like:

```
Children, extend type {@see ExpanderTitleBaseProps} or {@see ExpanderContentBaseProps}
```

Open issue for this.
https://github.com/microsoft/TypeScript/issues/35524

Reply:
That's a good idea, maybe we should test it with styleguidist and apply where necessary. Parts of our API documentation could use improvements.